src/main/kotlin/sh/hsp/techtree/FileSystemYamlReader.kt

```diff
@@ -5,15 +5,65 @@ import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import com.fasterxml.jackson.module.kotlin.readValue
 import com.fasterxml.jackson.module.kotlin.registerKotlinModule
 import java.io.File
+import java.io.IOException
 import java.net.URL
 
 class FileSystemYamlReader : YamlReader {
     override fun readToModel(resource: URL): TreeModel {
         val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
         val file = File(resource.toURI())
         val yamlData = file.readText()
+        val treeModel: TreeModel = mapper.readValue(yamlData)
 
-        return mapper.readValue(yamlData)
+        validateTreeModel(treeModel)
+
+        return treeModel
+    }
+
+    private fun validateTreeModel(treeModel: TreeModel) {
+        val titleList = treeModel.nodes.drop(1).map { it.title }
```

Comment:
Ten `drop(1)` nie do końca jest stabilny, bo wychodzi z założenia, że: 
1. Yaml ma strukturę drzewa z 1 rootem - dla takiego przykładu kod się wysypie:
```
nodes:
  - title: "miau"
    requires:
      - "kot"
      - "tygrys"
  - title: "kot"
  - title: "tygrys"
  - title: "hau"
    requires:
      - "wilk"
      - "pies"
  - title: "wilk"
  - title: "pies"
``` 

2. Node'y w yamlu są w odpowiedniej kolejności, dla takiego przykładu się wysypie:
```
  - title: "kot"
  - title: "tygrys"
  - title: "miau"
  requires:
    - "kot"
    - "tygrys"
```


Spodziewałem się tutaj raczej sprawdzenia czy w liście tytułów znajduje się lista requires czyli raczej coś takiego: `titleList.containsAll(requiresList)`

Reply:
Kurde mogłem się wypytać jak to dokładnie ma wyglądać xD. 
Zapamiętam na przyszłość